### PR TITLE
GORA-429 Implement Maven forbidden-apis plugin in Gora

### PR DESCRIPTION
**Pull request by lewismc**
Hi Folks, this is a PR for making Gora adhere to the forbidden api's checks. It turned out to be quite a large patch but all of the changes are minor.

### CODE REVIEW

File: gora-accumulo/src/main/java/org/apache/gora/accumulo/store/AccumuloStore.java [hunk 1]
```diff
@@ -248,7 +247,6 @@ public K fromBytes(Class<K> clazz, byte[] val) {
         avroEncoder.flush();
         return baos.toByteArray();
       } catch (IOException e) {
-        e.printStackTrace();
```
Comment (renato2099) [inline gora-accumulo/src/main/java/org/apache/gora/accumulo/store/AccumuloStore.java]: So why not printing the exception here? I mean at least we should print out a message saying that something went wrong, or am I missing anything?
Overall looks good to me @lewismc ! Thanks!

Comment (lewismc) [inline gora-accumulo/src/main/java/org/apache/gora/accumulo/store/AccumuloStore.java]: Hi @renato2099 actually, I think the answer is no. We should not be printing an exception here. I _think_ the logic may be flawed. 
We make a check in the code for the object being null (in which case we call firstNullSchemaTypeIndex and then process with the avro encoder) or the alternative which is that the object is not null (in which case we call firstNotNullSchemaTypeIndex and then process with the avro encoder).
This method is recursive. 
If you look at the very last else statement 
```
    } else {
      return toBytes(o);
    }
```
this throws no logging error or stack trace exception right? I think that this logic is OK. It was wrong in the first place. 
 * to throw an exception would mean that the recursive code could never be called.
 * to log an error would be useless and what other state can the object be in other than either 'null' or 'not null', in which case it has already been processed by the preceding statements!

wdyt. Thank you for review.

**Comment by renato2099** [comment]
Hi @lewismc 
I think the logic is fine, the non-null checks you are describing are for checking nullable schemas, and that is what you've described. Yes, it is recursive if there are nested objects, but the exception is for when "something" goes wrong with serialization (for example, class changes can easily make your serialized data unreadable if you were using reflection) that is why Java forces us to throw an IOException and not NullPointerException. And caching this exception makes us "retry" serialization calling again to <return toBytes(o)> on the object. If we don't print the IO exception, we would probably get an stackOverFlowException because of the recursion and catching this error would be a lot harder.

File: gora-accumulo/src/main/java/org/apache/gora/accumulo/store/AccumuloStore.java [hunk 2]
```diff
@@ -142,7 +142,7 @@ public Object fromBytes(Schema schema, byte data[]) throws GoraException {
           data = decoder.readBytes(null).array();
         }
       } catch (IOException e) {
-        e.printStackTrace();
+        LOG.error(e.getMessage());
```
Comment (renato2099) [inline gora-accumulo/src/main/java/org/apache/gora/accumulo/store/AccumuloStore.java]: LGTM!!! Thanks @lewismc !

**Pull request merged**